packages/core/lib/util/get-response.ts

```diff
@@ -0,0 +1,144 @@
+import { ExecutionResult } from "graphql";
+import { ExecutionPatchResult } from "../types";
+import { ReadableStream, Response } from "cross-undici-fetch";
+import { encodeString } from "./encode-string";
+
+export type TransformResultFn = (result: ExecutionResult | ExecutionPatchResult) => any;
+export const DEFAULT_TRANSFORM_RESULT_FN: TransformResultFn = (result: ExecutionResult) => result;
+
+export function getRegularResponse(executionResult: ExecutionResult, transformResult = DEFAULT_TRANSFORM_RESULT_FN): Response {
+  const transformedResult = transformResult(executionResult);
+  const responseBody = JSON.stringify(transformedResult);
+  const decodedString = encodeString(responseBody);
+  const headersInit: HeadersInit = {
+    "Content-Type": "application/json",
+    "Content-Length": decodedString.byteLength.toString(),
+  };
+  const responseInit: ResponseInit = {
+    headers: headersInit,
+    status: 200,
+  };
+  return new Response(decodedString, responseInit);
+}
+
+export function getMultipartResponse(
+  asyncExecutionResult: AsyncIterable<ExecutionPatchResult<any>>,
+  transformResult = DEFAULT_TRANSFORM_RESULT_FN
+): Response {
+  const headersInit: HeadersInit = {
+    Connection: "keep-alive",
+    "Content-Type": 'multipart/mixed; boundary="-"',
+    "Transfer-Encoding": "chunked",
+  };
+  const responseInit: ResponseInit = {
+    headers: headersInit,
+    status: 200,
+  };
+
+  let iterator: AsyncIterator<ExecutionResult<any>>;
+
+  const readableStream = new ReadableStream({
+    async start(controller) {
+      iterator = asyncExecutionResult[Symbol.asyncIterator]();
+      controller.enqueue(`---`);
+    },
+    async pull(controller) {
+      const { done, value } = await iterator.next();
+      if (done) {
+        queueMicrotask(() => {
+          controller.enqueue("\r\n-----\r\n");
+          controller.close();
+        });
+      } else {
+        const transformedResult = transformResult(value);
+        const chunk = JSON.stringify(transformedResult);
+        const encodedChunk = encodeString(chunk);
+        controller.enqueue('\r\n');
+        controller.enqueue('Content-Type: application/json; charset=utf-8\r\n');
+        controller.enqueue('Content-Length: ' + encodedChunk.byteLength + '\r\n');
```

Comment:
Again, absolutely 100% zero need or anything for this line to exist. The body isn't being inspect by any cdn or intermediary. 

I mean this line is effectively the same as returning json, and the cdn have a whinge coz there was no closing bracket. 

It's the body, and those things aren't snooped.

Reply:
As I already said above, this is not something this PR covers.